### DISCUSSION TITLE
Only one queryNode performs the loading job when the index_type is DISKANN.

### DISCUSSION THREAD

**Discussion by dragonborn-lxc** [milvus-io/milvus#35431] [Q&A and General discussion]
The queryNode and indexNode were configured with 2 replicas, but only one queryNode performed the loading job after executing collection.load(). Why?

**Comment by yhmo** [comment]
Could you provide more details about how you observed "only one queryNode performed the loading job after executing collection.load()".

**Reply by dragonborn-lxc** [reply]
I have added podAntiAffinity configuration to ensure that the queryNode pods will not be deployed on the same K8s node.
When the index_type is IVF_FLAT, I noticed the following log in two queryNode pods:
![image](https://github.com/user-attachments/assets/8e027b7c-44f4-469e-92da-e16285e4e5f2)
![image](https://github.com/user-attachments/assets/fc26cc53-642e-483a-b800-8ebbf0df6a49)
and I observed that the memory usage in each queryNode pod was increasing.
However, when the index_type is DISKANN, only one queryNode displayed the aforementioned log, and the index files were created in the this pod:
![image](https://github.com/user-attachments/assets/6025ae9c-9046-4eed-9e94-35e1f520b506)
but the index files were not generated in the other queryNode.
The image below shows the information for two queryNodes:
![image](https://github.com/user-attachments/assets/d3d260ea-0199-46ae-8e04-e0fb9e8f3871)

**Reply by dragonborn-lxc** [reply]
I found that the index files were randomly generated in either queryNode 1 or queryNode 2, but only in one of the queryNodes.
Both collections (IVF_FLAT and DISKANN) have identical structures and the same number of entities (500K).

**Comment by dragonborn-lxc** [comment]
Does queryNode only support one pod, or does multiple queryNodes only support deploying on a single K8s node?
